plugin/helpers_single_node.go

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package plugin
+
+import "errors"
+
+// RunOnSingleNode is a wrapper function which should guarantee that only one plugin instance can run a function associated with the given id at a time
+// The id parameter is an identifier that uses for synchronization must be unique between each function.
+func (p *HelpersImpl) RunOnSingleNode(id string, f func()) (bool, error) {
+	updated, err := p.KVCompareAndSetJSON(id, nil, true)
```

Comment:
Why does this need to be stored as JSON? We can just use plain bytes here and go with https://developers.mattermost.com/extend/plugins/server/reference/#API.KVCompareAndSet